stdlib/config/config.lua

```diff
@@ -0,0 +1,152 @@
+--- Config module
+-- @module Config
+
+require 'stdlib/core'
+require 'stdlib/string'
+require 'stdlib/table'
+
+-----------------------------------------------------------------------
+--Setup repeated code for use in sub functions here
+-----------------------------------------------------------------------
+local reservedCharacters = '`~!@#$%^&*+=|;:/\\\'",?()[]{}<>'
+local testReservedCharacters = function(path)
+    local reservedCharacters = reservedCharacters
+    for c in reservedCharacters:gmatch('.') do
+        if path:find(c, 1, true) then
+            return c
+        end
+    end
+    return nil
+end
+
+if not Config then Config = {} end
+Config = {}
+
+--- Creates a new Config object
+-- to ease the management of a config table.
+-- @param config_table [required] The table to be managed.
```

Comment:
Given the complexity of using config, usage examples seem very important here. See for an example of usage docs, https://github.com/Afforess/Factorio-Stdlib/blob/master/stdlib/area/area.lua#L108-L111

Multiple usage sections can be defined, to create multiple examples.